eth/beacon/helpers.py

```diff
@@ -362,3 +340,104 @@ def get_block_committees_info(parent_block: 'BaseBeaconBlock',
+        state,
+        slot,
+        epoch_length,
+    )
+    try:
+        first_shard_committee = shard_committees[0]
+    except IndexError:
+        raise ValueError("shard_committees should not be empty.")
+
+    proposer_committee_size = len(first_shard_committee.committee)
+
+    if proposer_committee_size <= 0:
+        raise ValueError(
+            "The first committee should not be empty"
+        )
+
+    return first_shard_committee.committee[slot % len(first_shard_committee.committee)]
+
+
+#
+# Bitfields
+#
+@to_tuple
+def get_attestation_participants(state: 'BeaconState',
+                                 slot: int,
+                                 shard: int,
+                                 participation_bitfield: bytes,
+                                 epoch_length: int) -> Iterable[int]:
+    """
+    Return the participant indices at for the ``slot`` of shard ``shard``
+    from ``participation_bitfield``.
+    """
+    # Find the relevant committee
+    # Filter by slot
+    shard_committees_at_slot = get_shard_committees_at_slot(
+        state,
+        slot,
+        epoch_length,
+    )
+    # Filter by shard
+    shard_committees = tuple(
+        [
+            shard_committee
+            for shard_committee in shard_committees_at_slot
+            if shard_committee.shard == shard
+        ]
+    )
+
+    try:
+        shard_committee = shard_committees[0]
+    except IndexError:
+        raise ValueError("shard_committees should not be empty.")
+
+    if len(participation_bitfield) != get_bitfield_length(len(shard_committee.committee)):
+        raise ValueError(
+            'Invalid bitfield length,'
+            "\texpected: %s, found: %s" % (
+                get_bitfield_length(len(shard_committee.committee)),
+                len(participation_bitfield),
+            )
+        )
+
+    # Find the participating attesters in the committee
+    for bitfield_index, validator_index in enumerate(shard_committee.committee):
+        if has_voted(participation_bitfield, bitfield_index):
+            yield validator_index
+
+
+#
+# Misc
+#
+def get_effective_balance(validator: 'ValidatorRecord', max_deposit: int) -> int:
+    """
+    Return the effective balance (also known as "balance at stake") for the ``validator``.
+    """
+    return min(validator.balance, max_deposit)
+
+
+def get_new_validator_registry_delta_chain_tip(current_validator_registry_delta_chain_tip: Hash32,
+                                               index: int,
+                                               pubkey: int,
+                                               flag: int) -> Hash32:
+    """
+    Compute the next hash in the validator registry delta hash chain.
+    """
+    return blake(
+        current_validator_registry_delta_chain_tip +
+        flag.to_bytes(1, 'big') +
+        index.to_bytes(3, 'big') +
+        # TODO: currently, we use 256-bit pubkey which is different form the spec
```

Comment:
Not for this PR but something I think we should address.

We have tended to use `eth_keys` data types for private and public key representations.  I suspect that we'll have to make some modifications to `eth_keys` to support new key types?  If this is accurate, do you think you can open issues both here and in `eth_keys` to track those needs?